test_common.go

```diff
@@ -174,20 +153,24 @@ func pubKeyFromJWK(t *testing.T, j string) crypto.PublicKey {
 	require.Nil(t, err)
 
 	var crv elliptic.Curve
+	var alg cose.Algorithm
 
 	switch v := key.(type) {
 	case *ecdsa.PrivateKey:
 		crv = v.Curve
-
 		if crv == elliptic.P256() {
-
+			alg = cose.AlgorithmES256
 			break
 		}
 		require.True(t, false, "unknown elliptic curve %v", crv)
 	default:
 		require.True(t, false, "unknown private key type %v", reflect.TypeOf(key))
 	}
+	return alg, key
+}
 
+func pubKeyFromJWK(t *testing.T, j string) crypto.PublicKey {
+	_, key := getAlgAndKeyFromJWK(t, j)
 	vk := key.Public()
 	return vk
```

Comment:
```suggestion
	return key.Public()
```

Reply:
I think I will leave it like this for ease of code readability!